libs/data-mapper/src/lib/components/configPanel/SelectExistingSchema.tsx

```diff
@@ -0,0 +1,124 @@
+};
+
+const useStyles = makeStyles({
+  option: {
+    ...shorthands.borderBottom('.5px', 'solid', tokens.colorNeutralStroke1),
+  },
+  combobox: {
+    width: '290px',
+  },
+  errorMessage: {
+    color: tokens.colorPaletteRedForeground1,
+  },
+});
+
+export const SelectExistingSchema = (props: SelectExistingSchemaProps) => {
+  const styles = useStyles();
+  // intl
+  const intl = useIntl();
+  const dropdownAriaLabel = intl.formatMessage({
+    defaultMessage: 'Select the schema for dropdown',
+    description: 'Schema dropdown aria label',
+  });
+  const schemaDropdownPlaceholder = useMemo(() => {
+    if (props.schemaType === SchemaType.Source) {
+      return intl.formatMessage({
+        defaultMessage: 'Select a source schema',
+        description: 'Source schema dropdown placeholder',
+      });
+    } else {
+      return intl.formatMessage({
+        defaultMessage: 'Select a target schema',
+        description: 'Target schema dropdown placeholder',
+      });
+    }
+  }, [intl, props.schemaType]);
+
+  const availableSchemaList = useSelector((state: RootState) => state.schema.availableSchemas);
+
+  const dataMapDropdownOptions = useMemo(() => availableSchemaList ?? [], [availableSchemaList]);
+
+  const [matchingOptions, setMatchingOptions] = React.useState([...dataMapDropdownOptions]);
+
+  const setSelectedSchema = props.setSelectedSchema;
+
+  const onSelectOption = useCallback(
+    (option?: string) => {
+      setSelectedSchema(option);
+    },
+    [setSelectedSchema]
+  );
+
+  const updateOptions: ComboboxProps['onChange'] = (event) => {
+    const value = event.target.value.trim();
+    if (value && value.length > 0) {
+      const matches = dataMapDropdownOptions.filter((o) => o.toLowerCase().indexOf(value?.toLowerCase()) === 0);
+      setMatchingOptions(matches);
+    }
+    if (value.length === 0) {
+      setMatchingOptions(dataMapDropdownOptions);
+    }
+    onSelectOption(value);
+  };
+
+  const sortedOptions: string[] = matchingOptions;
+  const formattedOptions =
+    sortedOptions.length !== 0
+      ? sortedOptions.map((option) => optionWithPath(option))
+      : dataMapDropdownOptions.map((option) => optionWithPath(option));
+
+  return (
+    <>
+      <Text className={styles.errorMessage}>{props.errorMessage}</Text>
+      <Combobox
+        aria-label={dropdownAriaLabel}
+        placeholder={schemaDropdownPlaceholder}
+        onChange={updateOptions}
+        onOptionSelect={(e, data) => props.setSelectedSchema(data.optionValue)}
+        freeform={true}
+        autoComplete="on"
+        className={styles.combobox}
+        appearance="outline"
+      >
+        {formattedOptions}
+      </Combobox>
+    </>
+  );
+};
+
+const optionWithPath: React.FC<string> = (option: string) => {
+  const [fileName, filePath] = getFileNameAndPath(option);
+
+  // styling
+  const stackTokens: IStackTokens = {
+    childrenGap: 5,
+  };
+
+  // danielle fix
```

Comment:
Still need this?

Reply:
removed! error was from not capitalizing first letter of React component oops